client/app/src/main/java/com/example/criengine/Activities/RequestActivity.java

```diff
@@ -24,40 +28,39 @@
 public class RequestActivity extends AppCompatActivity {
     BorrowerBooksListAdapter borrowerBooksListAdapter;
     ArrayList<Book> borrowerBooks;
-    DatabaseWrapper dbw = DatabaseWrapper.getWrapper();
+    DatabaseWrapper dbw;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_request);
 
-        // TODO: get borrowed or requested books from db:
-        // borrowerBooks = dbw.getMyProfile().getbooksBorrowedOrRequested();
-        Map<String, Object> testData = new HashMap<String, Object>(){{
-                put("bookID", "123");
-                put("owner", "JohnDoe");
-                put("title", getString(R.string.lorem_ipsum));
-                put("author", "JaneDoe");
-                put("description", "A book");
-                put("isbn", "isbn");
-                put("status", "available");
-                put("requesters", new ArrayList<String>());
-        }};
+        dbw = DatabaseWrapper.getWrapper();
 
-        Book book1 = new Book("John Doe", getString(R.string.lorem_ipsum), "JaneDoe","A book", "isbn","available");
-        Book book2 = new Book("John Doe", getString(R.string.lorem_ipsum), "JaneDoe","A book", "isbn","available");
-        Book book3 = new Book("John Doe", getString(R.string.lorem_ipsum), "JaneDoe","A book", "isbn","available");
-        Book book4 = new Book("John Doe", getString(R.string.lorem_ipsum), "JaneDoe","A book", "isbn","available");
-        Book[] books = {
-                book1, book2, book3, book4
-        };
         borrowerBooks = new ArrayList<Book>();
-        borrowerBooks.addAll(Arrays.asList(books));
+//        borrowerBooks.addAll(Arrays.asList(books));
         borrowerBooksListAdapter = new BorrowerBooksListAdapter(this, borrowerBooks);
 
         ListView bookNameTextView = findViewById(R.id.bookListView);
         bookNameTextView.setAdapter(borrowerBooksListAdapter);
 
+        dbw.getProfile(dbw.userId).addOnSuccessListener(
```

Comment:
Oh totally possible. I'm still in the camp of "we should always have a profile object around" though since stuff would load twice as fast if we did. If we want to use/cache the profile then this is how I plan to do it but if not then I'll clean it up just like how you suggest

Reply:
Okay, I see. So this has to be done with a Task and onSuccessListener in order to get the latest BorrowedOrRequestedBooks array. Okie! in that case, this looks good 👍